В общем я не прочел инструкцию по сдаче проекта. Сразу залил од в реп…

DenisSangi opened the pull request:
…озитроий без ПРа. Теперь добавляю ПР, просто внес комменты во все файлы.

Commit:
В общем я не прочел инструкцию по сдаче проекта. Сразу залил од в репозитроий без ПРа. Теперь добавляю ПР, просто внес комменты во все файлы.

nrasskosova commented on src/main/java/Account.java:
```diff
@@ -1,3 +1,4 @@
+//Просто добавлю комментарий чтобы были изменения для Пулл Реквеста
```
я могу добавить комментарий в этом место, а не по всему коду. Нужно было сделать новый pr со всеми изменениями.

1. Проверку делаем не if конструкциями, а через логическую операцию &&
2. Нет проверки на кол-во пробелов

Commit:
Доработки:
1. Заменил If на логические операторы &&;
2. Добавил метод и тесты на проверку количества пробелов;

Непонятки:
Почему-то тест nameEndsWithSpaceTest() постоянно ловит баг, хотя условие составлено верно. Аналогичный nameStartsWithSpaceTest() проходит норм.

nrasskosova commented on src/main/java/Account.java:
```diff
@@ -11,21 +15,11 @@ public boolean checkNameToEmboss() {
+        return !(name.length() < 3) && !(name.length() > 19) && name.contains(" ") && !(name.startsWith(" ") && !(name.endsWith(" ")));
+    }
+
+    public boolean onlyOneSpace(String input, String subStr) {
```
Давай вместо этого метода сделаем метод, который проверяет, что кол-во пробелов в строке = 1 и оно не в начале и не в конце. Кол-во пробелов можно проверить с помощью цикла или стримов, с циклами - легче, со стримами - короче.
И этот новый метод будем вызывать в методе checkNameToEmboss сразу, а не из тестов.

nrasskosova commented on src/test/java/AccountTest.java:
```diff
@@ -38,16 +39,16 @@ public void toLongNameTest(){
     @Test
     @DisplayName("Checking if space is a last symbol")
     @Description("Basic negative test when name ends with space")
     public void nameEndsWithSpaceTest(){
```
этот тест падает

nrasskosova commented on src/test/java/AccountTest.java:
```diff
@@ -56,7 +57,24 @@ public void nameEndsWithSpaceTest(){
+    @Test
+    @DisplayName("Checking if there is only one space")
+    @Description("Basic negative test when name didn't includes any space")
+    public void onlySingeSpaceTestPositive(){
```
дублирование теста.  Должна быть одна проверка, которая покрывает одновременно все 5 условий

nrasskosova commented on src/test/java/AccountTest.java:
```diff
@@ -56,7 +57,24 @@ public void nameEndsWithSpaceTest(){
+    @Test
+    @DisplayName("Checking if there is more than one space")
+    @Description("Basic negative test when name didn't includes any space")
+    public void onlySingeSpaceTestNegative(){
```
Если сделать входную строку больше 19 символов и с 2 параметрами при вызове этого метода проверки ошибки не будет.

Commit:
Доработки:
1. В Класс Account добавил метод isSingleSpace().
Метод проверяет количество пробелов и возвращает true только если пробелов = 1, и это не первый/последний символ.
Наворотил ужасное дерево условий везде, но почему то в более красивом виде теты показывают неверный результат. Так и не понял почему.

P.S.:
Ревьюер поставь зачёт.
Уже диплом начался не хочу инвестировать больше времени в проект 4-го спринта.
Тем более этот спринт больше про инсрументы CI/CD и и настройки.

Commit:
Доработки:
1. Добавил тест на гриничное значение 19 символов.

nrasskosova commented on src/main/java/Account.java:
```diff
@@ -11,21 +12,19 @@ public boolean checkNameToEmboss() {
-            return false;
-        } else if (name.contains(" ")) {
-            return true;
+        if (name.length() > 2 && name.length() <= 19)  if (isSingleSpace(name)) return true;
```
получилось снова замудрено.
давай сделаем так:
 return name.length()>=3
                && name.length() <=19
                && !name.startsWith(" ")
                && !name.endsWith(" ")
                && isSingleSpace(name);

Когда необходимо, чтоб выполнялись все условия, то такие условия нужно объединить с помощью логического И

nrasskosova commented on src/main/java/Account.java:
```diff
@@ -11,21 +12,19 @@ public boolean checkNameToEmboss() {
+        return false;
+    }
+
+    public boolean isSingleSpace(String name){
```
по названию этот метод должен только проверить наличие одного пробела в строке. Но он еще проверяет и позицию пробела в строке. Либо нужно изменить метод, либо его переименовать

nrasskosova commented on src/test/java/AccountTest.java:
```diff
@@ -3,6 +3,7 @@
 import org.junit.Test;
 
+//Просто добавлю комментарий чтобы были изменения для Пулл Реквеста
 public class AccountTest {
```
Эти тесты дублируются, Хорошо бы ихх параметризировать и вынести в параметры строку и булевый результат проверки.
Нет проверок на пустую строку и на минимальный размер - "D S"

Commit:
Доработки:
1. Изменил метод checkNameToEmboss;
2. Изменил метод isSingleSpace;
3. обавил проверки на 3 символа и пустую строку;
4. Параметризовал всё это дело как царь.

Commit:
Доработки в догонку:
1. Удалил неиспользуемые импорты.

Commit:
Доработки в догонку - 2:
1. Удалил какую-то непонятную аннтоацию @Contract перед параметризованными тестами.
